examples/chip-tool/commands/payload/AttestationResponseParser.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "AttestationResponseParser.h"
+#include <credentials/CertificationDeclaration.h>
+#include <string>
+
+using namespace ::chip;
+
+enum
+{
+    kTag_CertDecleration = 1,
+    kTag_Signature       = 2,
+};
+
+CHIP_ERROR AttestationResponseParser::Run()
+{
+    ChipLogProgress(chipTool, "Parsing Attestation Response.");
+    ParseAttestationResponse(mAttResponse);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AttestationResponseParser::ParseAttestationResponse(chip::ByteSpan & responsePayload)
+{
+    chip::Credentials::CertificationElements certElements;
+    chip::ByteSpan certDecleration;
+    chip::ByteSpan signature;
+    chip::ByteSpan certDeclerationElements;
+    chip::TLV::TLVReader reader;
+    chip::TLV::TLVType outerContainer1;
+    size_t indent = 0;
+
+    reader.Init(responsePayload);
+    ReturnErrorOnFailure(reader.Next(chip::TLV::kTLVType_Structure, chip::TLV::AnonymousTag()));
+    ReturnErrorOnFailure(reader.EnterContainer(outerContainer1));
+
+    ReturnErrorOnFailure(reader.Next(chip::TLV::ContextTag(kTag_CertDecleration)));
+    ReturnErrorOnFailure(reader.Get(certDecleration));
+
+    ReturnErrorOnFailure(reader.Next(chip::TLV::ContextTag(kTag_Signature)));
+    ReturnErrorOnFailure(reader.Get(signature));
+
+    ReturnErrorOnFailure(chip::Credentials::CMS_ExtractCDContent(certDecleration, certDeclerationElements));
```

Comment:
So "certDeclerationElements" is what the spec calls `certification-elements`?

Reply:
certDeclerationElements is actually the cd_content which contains certification-elements. I will updated the name of certDeclerationElements to certDeclarationContent.

Reply:
Updated